src/core/metrics/app-metrics.ts

```diff
@@ -71,11 +73,13 @@ export class AppMetrics {
     this.listener = globalThis.addEventListener(METRICS_EVENT, ((e: MetricsEvent) => {
       this.processEvent(e.detail.name, e.detail.payload);
     }) as EventListener);
+    currentModeAtom.subscribe((mode) => (this.mode = mode));
```

Comment:
Cool! One more thing - whenever we have subscribe, we also need cleanup\unsubscribe mechanism

Reply:
cleanup implemented